### PR TITLE
chore: reorganize app codebase

### PR DESCRIPTION
**Pull request by oliverlaz**
With this PR, few code readability improvements are done:
- hooks are now isolated into their own individual modules for easier navigation and understanding
- app dependencies are now injected as props.
- Giphy integration is isolated as well as part of this change
- StreamChat client is not a singleton anymore because of issues with improper cleanup (see: https://github.com/GetStream/stream-chat-react/issues/1616)

### CODE REVIEW

File: social-messenger-ts/src/hooks/useConnectUser.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      await client.connectUser(userToConnect, userToken);
+      setChatClient(client);
+    };
+
+    connectUser().catch((e) => {
+      console.error(`Failed to connect user`, e);
+    });
+
+    return () => {
+      chatClient?.disconnectUser().catch((e) => {
```
Comment (arnautov-anton) [inline social-messenger-ts/src/hooks/useConnectUser.ts:33]: I'd rather use `client` instance created within this hook's closure that way we can be sure 100% it'll always calls for disconnect.

File: social-messenger-ts/src/hooks/useConnectUser.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+import { StreamChat } from 'stream-chat';
+
```
[the remark this answers is not in the document]
Comment (arnautov-anton) [reply social-messenger-ts/src/hooks/useConnectUser.ts:3]: First SDK PR, @oliverlaz? *wink, wink*

File: social-messenger-ts/src/hooks/useMobileView.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const [isMobileNavVisible, setMobileNav] = useState(false);
+  useEffect(() => {
+    const mobileChannelList = document.querySelector('#mobile-channel-list');
+    if (isMobileNavVisible && mobileChannelList) {
+      mobileChannelList.classList.add('show');
+      document.body.style.overflow = 'hidden';
+    } else if (!isMobileNavVisible && mobileChannelList) {
+      mobileChannelList.classList.remove('show');
+      document.body.style.overflow = 'auto';
+    }
```
Comment (arnautov-anton) [inline social-messenger-ts/src/hooks/useMobileView.ts:16]: This feels weird though I'm not sure if it's worth refactoring - I think CSS variables would be a better solution for this.

File: social-messenger-ts/src/hooks/useConnectUser.ts (hunk 1 above)
Comment (oliverlaz) [reply social-messenger-ts/src/hooks/useConnectUser.ts:33]: Good point, I'll update the hook right away.

File: social-messenger-ts/src/hooks/useUpdateAppHeightOnResize.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ */
+export const useUpdateAppHeightOnResize = () => {
+  useEffect(() => {
+    /*
+     * Get the actual rendered window height to set the container size properly.
+     * In some browsers (like Safari) the nav bar can override the app.
+     */
+    const setAppHeight = () => {
+      const doc = document.documentElement;
+      doc.style.setProperty('--app-height', `${window.innerHeight}px`);
```
Comment (petyosi) [inline social-messenger-ts/src/hooks/useUpdateAppHeightOnResize.ts:15]: Idea for future, we can probably use one of [the relative units for that](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#numbers_lengths_and_percentages).

**Review by petyosi — APPROVED** [review]

**Review by MartinCupela — APPROVED** [review]

File: social-messenger-ts/src/hooks/useConnectUser.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    const client = new StreamChat<SCG>(apiKey, {
+      enableInsights: true,
+      enableWSFallback: true,
+    });
+
+    // Under some circumstances, a "connectUser" operation might be interrupted
+    // (fast user switching, react strict-mode in dev). With this flag, we control
+    // whether a "disconnectUser" operation has been requested before we
+    // provide a new StreamChat instance to the consumers of this hook.
+    let didUserConnectInterrupt = false;
```
Comment (arnautov-anton) [inline social-messenger-ts/src/hooks/useConnectUser.ts:35]: Way better naming, I like it! :ok_hand:

**Review by arnautov-anton — APPROVED** [review]

**Pull request merged**